fsspec/implementations/dbfs.py

```diff
@@ -70,7 +90,16 @@ def ls(self, path, detail=True, **kwargs):
             but also additional information on file sizes
             and types.
         """
-        out = self._ls_from_cache(path)
+        try:
+            out = self._ls_from_cache(path)
+        except FileNotFoundError:
+            # This happens if the `path`'s parent was cached, but `path` is not
+            # there. This suggests that `path` is new since the parent was
+            # cached. Attempt to invalidate parent's cache before continuing.
+            with suppress(Exception):
+                del self.dircache[self._parent(path)]
```

Comment:
```suggestion
            self.dircache.pop(self._parent(path))
```

Reply:
I think `pop` is perfectly descriptive of the operation (better than a context and indent), and you need not assign it to the local variable `_` :)